async_service/asyncio.py

```diff
@@ -246,10 +247,11 @@ def run_task(
         current_task = asyncio.Task.current_task()
         if not _root and current_task not in self._service_task_dag:
             raise Exception(f"TODO: unknown task {current_task}")
+        if name is None:
+            name = repr(async_fn)
+
         task = asyncio.ensure_future(
-            self._run_and_manage_task(
-                async_fn, *args, daemon=daemon, name=name or repr(async_fn)
```

Comment:
I'm curious as to how we were always using the function's repr even when a name was passed, as you say in the PR's description. This ought to have used the name when it was not None or the empty string, no?

Reply:
I realize I was actually wrong about this, though I like the new logic more.  I was trying to debug something, provided an explicit name, searched the logs and didn't find it, but found a bunch of <couroutine ....> names instead, and jumped to the conclusion that the name wasn't being used.

I still like this cleanup of making the logic a bit more explicit.